frappe/__init__.py

```diff
@@ -1142,6 +1143,24 @@ def get_cached_value(
 	return values
 
 
+_BaseDocument = TypeVar("_BaseDocument", bound="BaseDocument")
+
+
+@overload
+def get_doc(document: "_BaseDocument", /) -> "_BaseDocument":
+	pass
+
+
+@overload
+def get_doc(self, doctype: str, name: str | None, *, for_update: bool | None = None) -> "Document":
+	pass
+
+
+@overload
+def get_doc(self, document: DocumentDict, /) -> "Document":
```

Comment:
document here can also be _splatted_. i.e. `frappe.get_doc(a=b, c=d, x=y)`

Not sure if it's possible to represent that too. :thinking:

Reply:
Quite complicated: https://stackoverflow.com/questions/37031928/type-annotations-for-args-and-kwargs

Reply:
Silly me, that should've been obvious.
In the splatted form we simply use `(doctype:str, name:str|None, **kwargs)`
I'll push it in a moment.

So there are two functions here, one that creates, other that retrieves.
- creating accepts doctype, optional name and fields, either as dict or kwargs
- retrieving accepts doctype, name required except for singles, thus optional, and optional for_update.
Is this correct?

What DX stands for?

BTW, since there's new_doc, may I suggest deprecating the creating behaviour of get_doc and mark it for removal in V16?